Assets/HoloToolkit/Input/Scripts/InputEventData/BaseInputEventData.cs

```diff
@@ -23,18 +23,24 @@ public abstract class BaseInputEventData : BaseEventData, IInputSourceInfoProvid
         /// <summary>
         /// An optional, input-source-dependent object to be associated with this event.
         /// </summary>
-        public object Tag { get; private set; }
+        public object[] Tag { get; private set; }
+
+        /// <summary>
+        /// An optional, input-source-independent tag to be associated with this event.
+        /// </summary>
+        public string EventOrigin { get; private set; }
 
         public BaseInputEventData(EventSystem eventSystem) : base(eventSystem)
         {
         }
 
-        protected virtual void BaseInitialize(IInputSource inputSource, uint sourceId, object tag)
+        protected virtual void BaseInitialize(IInputSource inputSource, uint sourceId, string eventOrigin, object[] tag = null)
```

Comment:
Still not a fan of this string.

Reply:
I almost feel like the source and it's id should be enough to determine what the events origin is.

Reply:
It's enough to differentiate between a single device, but not between multiple sources that use a single device, which is a common situation. (You agreed that this was useful at the the meeting where it was presented so I'm going to hold you to past-Stephen's opinion.)

This is definitely something that would have to survive the transition to #1603, assuming we close this branch.

Reply:
I think with my refactor work, because each Input Source is unique, you should no longer get events that aren't from a single source. 

Please review and let me know if that's not the case.

Reply:
It doesn't look like that's the case. Here's another way to tell: If I have 3 pointer components attached to my left controller, and I receive a click event from the InputSource associated with that controller, can your event tell me which of those 3 pointers was responsible for raising that click event?

Reply:
Yes, that is correct